examples/stabilizer_code.py

```diff
@@ -0,0 +1,340 @@
+    ranges of rows and columns. The matrix elements are integers {0, 1} interpreted as elements
+    of GF(2).
+
+    In short, this is the implementation of section 4.1 of the thesis.
+
+    Args:
+        M: The input/output matrix
+        min_row: The minimum row (inclusive) where the perform the elimination.
+        max_row: The maximum row (exclusive) where the perform the elimination.
+        min_col: The minimum column (inclusive) where the perform the elimination.
+        max_col: The maximum column (exclusive) where the perform the elimination.
+
+    Returns:
+        The rank of the matrix.
+    """
+    assert M.shape[1] % 2 == 0
+    n = M.shape[1] // 2
+
+    max_rank = min(max_row - min_row, max_col - min_col)
+
+    rank = 0
+    for r in range(max_rank):
+        i = min_row + r
+        j = min_col + r
+        pivot_rows, pivot_cols = np.nonzero(M[i:max_row, j:max_col])
+
+        if pivot_rows.size == 0:
+            break
+
+        pi = pivot_rows[0]
+        pj = pivot_cols[0]
+
+        # Swap the rows:
+        M[[i, i + pi]] = M[[i + pi, i]]
+
+        # Swap the columns:
+        M[:, [(j + pj), j]] = M[:, [j, (j + pj)]]
+
+        # Since the columns in the left and right half of the matrix represent the same qubit, we
+        # also need to swap the corresponding column in the other half.
+        j_other_half = (j + n) % (2 * n)
+        M[:, [(j_other_half + pj), j_other_half]] = M[:, [j_other_half, (j_other_half + pj)]]
+
+        # Do the elimination.
+        for k in range(i + 1, max_row):
+            if M[k, j] == 1:
+                M[k, :] = np.mod(M[i, :] + M[k, :], 2)
+
+        rank += 1
+
+    # Backward replacing to get identity
+    for r in reversed(range(rank)):
+        i = min_row + r
+        j = min_col + r
+
+        # Do the elimination.
+        for k in reversed(range(min_row, i)):
+            if M[k, j] == 1:
+                M[k, :] = np.mod(M[i, :] + M[k, :], 2)
+
+    return rank
+
+
+def _transfer_to_standard_form(
+    M: np.array, n: int, k: int
+) -> Tuple[np.array, np.array, np.array, int]:
+    """
+    Puts the stabilizer matrix in its standardized form, as in section 4.1 of the thesis.
+
+    Args:
+        M: The stabilizier matrix, to be standardized.
+        n: Dimension of the code words.
+        k: Dimension of the message words.
+
+    Returns:
+        The standardized matrix.
+        The logical Xs.
+        The logical Zs.
+        The rank of the matrix.
+    """
+
+    # Performing the Gaussian elimination as in section 4.1
+    r: int = _GaussianElimination(M, 0, n - k, 0, n)
+    _ = _GaussianElimination(M, r, n - k, n + r, 2 * n)
+
+    # Get matrix sub-components, as per equation 4.3:
+    # A1 = M[0:r, r : (n - k)]
+    A2 = M[0:r, (n - k) : n]
+    # B = M[0:r, n : (n + r)]
+    C1 = M[0:r, (n + r) : (2 * n - k)]
+    C2 = M[0:r, (2 * n - k) : (2 * n)]
+    # D = M[r : (n - k), n : (n + r)]
```

Comment:
Let's remove unused code rather than commenting it out.

Reply:
Done.